crates/chain/src/tx_graph.rs

```diff
@@ -762,107 +763,190 @@ impl<A: Anchor> TxGraph<A> {
+                },
+            )
+            .filter_map(Result::transpose)
     }
 
-    /// List outputs that are in `chain` with `chain_tip`.
+    /// Get a filtered list of outputs from the given `outpoints` that are in `chain` with
+    /// `chain_tip`.
     ///
-    /// This is the infallible version of [`try_list_chain_txouts`].
+    /// This is the infallible version of [`try_filter_chain_txouts`].
     ///
-    /// [`try_list_chain_txouts`]: Self::try_list_chain_txouts
-    pub fn list_chain_txouts<'a, C: ChainOracle<Error = Infallible> + 'a>(
+    /// [`try_filter_chain_txouts`]: Self::try_filter_chain_txouts
+    pub fn filter_chain_txouts<'a, C: ChainOracle<Error = Infallible> + 'a, S: Clone + 'a>(
         &'a self,
         chain: &'a C,
         chain_tip: BlockId,
-    ) -> impl Iterator<Item = FullTxOut<ObservedAs<A>>> + 'a {
-        self.try_list_chain_txouts(chain, chain_tip)
-            .map(|r| r.expect("error in infallible"))
+        outpoints: impl IntoIterator<Item = (S, OutPoint)> + 'a,
+    ) -> impl Iterator<Item = (S, FullTxOut<ObservedAs<A>>)> + 'a {
+        self.try_filter_chain_txouts(chain, chain_tip, outpoints)
+            .map(|r| r.expect("oracle is infallible"))
     }
 
-    /// List unspent outputs (UTXOs) that are in `chain` with `chain_tip`.
+    /// Get a filtered list of unspent outputs (UTXOs) from the given `outpoints` that are in
+    /// `chain` with `chain_tip`.
+    ///
+    /// `outpoints` is a list of outpoints we are interested in, coupled with the associated txout's
+    /// script pubkey index (`S`).
     ///
-    /// Floating outputs are not iterated over.
+    /// Floating outputs are ignored.
     ///
     /// # Error
     ///
-    /// An item can be an error if the [`ChainOracle`] implementation fails. If the oracle is
-    /// infallible, [`list_chain_unspents`] can be used instead.
+    /// An [`Iterator::Item`] can be an [`Err`] if the [`ChainOracle`] implementation (`chain`)
+    /// fails.
     ///
-    /// [`list_chain_unspents`]: Self::list_chain_unspents
-    pub fn try_list_chain_unspents<'a, C: ChainOracle + 'a>(
+    /// If the [`ChainOracle`] implementation is infallible, [`filter_chain_unspents`] can be used
+    /// instead.
+    ///
+    /// [`filter_chain_unspents`]: Self::filter_chain_unspents
+    pub fn try_filter_chain_unspents<'a, C: ChainOracle + 'a, S: Clone + 'a>(
         &'a self,
         chain: &'a C,
         chain_tip: BlockId,
-    ) -> impl Iterator<Item = Result<FullTxOut<ObservedAs<A>>, C::Error>> + 'a {
-        self.try_list_chain_txouts(chain, chain_tip)
-            .filter(|r| matches!(r, Ok(txo) if txo.spent_by.is_none()))
+        outpoints: impl IntoIterator<Item = (S, OutPoint)> + 'a,
+    ) -> impl Iterator<Item = Result<(S, FullTxOut<ObservedAs<A>>), C::Error>> + 'a {
+        self.try_filter_chain_txouts(chain, chain_tip, outpoints)
+            .filter(|r| !matches!(r, Ok((_, full_txo)) if full_txo.spent_by.is_some()))
```

Comment:
no.

Reply:
Clippy will complain otherwise 🫤

Reply:
Jokes, there is a way of doing it that improves readability by alot.

Reply:
https://github.com/bitcoindevkit/bdk/blob/fb75aa94a9f47d53ce7246c14d739fc53dc38197/crates/chain/src/tx_graph.rs#L867-L873